Prog/preclean-DC.Rmd

```diff
@@ -0,0 +1,184 @@
+---
+title: "Regional Housing Framework"
+subtitle: "Pre-clean DC parcel base data"
+author: ""
+output:
+  html_document:
+    number_sections: FALSE
+    self_contained: TRUE
+    code_folding: show
+    toc: TRUE
+    toc_float: TRUE
+    css: ../www/web_report.css
+    editor_options:
+      chunk_output_type: console
+---
+
+<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato" />
+
+Library: RegHsg
+
+Project: Regional Housing Framework
+
+Author: Yipeng adapted from Sarah Strochak
+
+Version: R 3.5.1, RStudio 1.1.423
+
+Last updated `r format(Sys.time(), '%B %d, %Y')`
+
+Environment: Local Windows session (desktop)
+
+
+```{r rmarkdown-setup, echo = FALSE}
+knitr::opts_chunk$set(warning = FALSE)
+knitr::opts_chunk$set(message = FALSE)
+# knitr::opts_chunk$set(eval = FALSE)
+```
+
+## Description
+
+The purpose of this program is to obtain additional data from the county to supplement the Black Knight data when possible.
+
+## Set-up
+Load libraries and functions
+```{r setup}
+library(tidyverse)
+
+source("../Macros/read-bk.R")
+source("../Macros/filter-bk.R")
+source("../Macros/select-vars.R")
+
+```
+
+
+Set FIPS code, filepath name, and directory for data storage (on L drive)
+
+```{r fips}
+currentfips <- "11001"
+filepath <- "DC"
+jdir <- paste0("L:/Libraries/RegHsg/Data/", filepath, "/")
+rdir <- paste0("L:/Libraries/RegHsg/Raw/", filepath, "/")
+
+```
+
+
+Create directory for data exports
+```{r dir}
+if (!dir.exists("../Data")) {
+  dir.create("../Data")
+}
+
+if (!dir.exists(paste0("L:/Libraries/RegHsg/Raw/", filepath))) {
+  dir.create(paste0("L:/Libraries/RegHsg/Raw/", filepath))
+}
+
+
+```
+
+Load in Black Knight data for the region, select jurisdiction and standard variables
+```{r read}
+if (!exists("region")) {
+  region <- read_bk("dc-cog-assessment_20181228.csv")
+} else {
+  warning("region data already read in")
+}
+
+jur <- region %>% 
+  filter_bk(fips = currentfips) %>% 
+  select_vars()
+
+```
+## Import DC files
+
+I used the DC parcel base data for merging to provide alternative property and lot information for cleaning DC data
+
+```{r import sas data}
+
+library(haven)
+
+parcelbase <- haven::read_sas("L:/Libraries/realprop/data/parcel_base.sas7bdat")
+
```

Comment:
please import parcel_geo and the x & y coordinates from there. Merge parcel geo with parcel base then merge with black nknight